### DISCUSSION TITLE
BMP预合成素材卡顿

### DISCUSSION THREAD

**Discussion by XinPolaris** [Tencent/libpag#2273] [Q&A] [answered]
【版本信息】
implementation 'com.tencent.tav:libpag:4.3.47' | Andorid 6.0 

【平台信息】
Android 原生，设备分辨率1280x480

【预期的表现】
动画流畅

【实际的情况】
动画卡顿严重。实测在性能高的机器上流畅度有提升，查阅了官网关于BMP预合成的优化方案，没有找到答案。请问这个动画有优化空间吗？除了调整分辨率还有别的手段吗？
另外，实际测试直接使用帧动画是流畅的。

【Demo及附件】

xml：
```
<org.libpag.PAGImageView
        android:id="@+id/pagView"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent" />
```
Activity：
```
binding.pagView.path = "assets://icon_steam_cycle_bmp.pag"
binding.pagView.setRepeatCount(-1)
binding.pagView.setScaleMode(PAGScaleMode.LetterBox)
binding.pagView.play()
```

素材：
[icon_steam_cycle_bmp.zip](https://github.com/Tencent/libpag/files/15148428/icon_steam_cycle_bmp.zip)

**Comment by kevingpqi123** [comment, marked as answer]
https://pag.art/docs/optimization.html
这篇文章讲的很清楚哈，如果追求性能，尽可能采用纯矢量导出，BMP 预合成仅仅是保底方案
如你这边的附件，首先检查下在 BMP 预合成上是否使用了其它的 AE 特性，这里是整体导出的，整体选择即可，从桌面预览工具上看，性能是偏差的
<img width="940" alt="截屏2024-05-01 11 43 44" src="https://github.com/Tencent/libpag/assets/16609221/e22924ab-0b8b-46c0-a167-090b39e34e58">
其次就是调整分辨率
